dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py

```diff
@@ -0,0 +1,304 @@
+        else:
+            endpoint = self.url + "/" + path
+        return endpoint
+
+    def clean_unsupported_metadata(self, component):
+        """
+        Convert the data type to string in case of `boolean` and `float`.
+        Currently we get `float` values for `Hosts`
+        :param component: metadata with unsupported data types
+        :return: metadata with supported data types
+        """
+        for key in component.keys():
+            if type(component[key]) is float:
+                component[key] = str(component[key])
+            elif type(component[key]) is bool:
+                component[key] = str(component[key])
+        return component
+
+    def get_host_identifiers(self, component):
+        host_identifiers = []
+        if component.get("esxiHostName"):
+            host_identifiers.append(Identifiers.create_host_identifier(component.get("esxiHostName")))
+        if component.get("oneAgentCustomHostName"):
+            host_identifiers.append(Identifiers.create_host_identifier(component.get("oneAgentCustomHostName")))
+        if component.get("azureHostNames"):
+            host_identifiers.append(Identifiers.create_host_identifier(component.get("azureHostNames")))
+        if component.get("publicHostName"):
+            host_identifiers.append(Identifiers.create_host_identifier(component.get("publicHostName")))
+        if component.get("localHostName"):
+            host_identifiers.append(Identifiers.create_host_identifier(component.get("localHostName")))
+        host_identifiers.append(Identifiers.create_host_identifier(component.get("displayName")))
+        return host_identifiers
+
+    def collect_topology(self, response, component_type):
+        """
+        Process each component type and map those with specific data
+        :param response: Response of each component type endpoint
+        :param component_type: Component type
+        :return: create the component on stackstate API
+        """
+        if "error" in response:
+            raise DynatraceError(response["error"].get("message"), response["error"].get("code"), component_type)
+        for item in response:
+            item = self.clean_unsupported_metadata(item)
+            data = dict()
+            external_id = item["entityId"]
+            identifiers = [Identifiers.create_custom_identifier("dynatrace", external_id)]
+            if component_type == "host":
+                host_identifiers = self.get_host_identifiers(item)
+                identifiers.extend(host_identifiers)
+            # derive useful labels and get labels from dynatrace tags
+            labels = self.get_labels(item)
+            data.update(item)
+            self.filter_item_topology_data(data)
+            data.update({
+                "identifiers": identifiers,
+                "tags": self.tags,
+                "domain": self.domain,
+                "environment": self.environment,
+                "instance": self.url,
```

Comment:
A comment here that his url is used in StackState for component action may be helpful